packages/server-wallet/src/protocols/state.ts

```diff
@@ -50,47 +49,54 @@ type SignedByMe = {latestSignedByMe: SignedStateWithHash};
 export type ChannelStateWithMe = ChannelState & SignedByMe;
 export type ChannelStateWithSupported = ChannelState & SignedByMe & WithSupported;
 
-export type Stage = 'Missing' | 'PrefundSetup' | 'PostfundSetup' | 'Running' | 'Final';
 /**
- *
- * @param state: SignedStateWithHash | undefined
- * Returns stage of state.
- *
- * Useful for partitioning the protocol state to decide what action to next take.
+ * The definition of a ChannelStatus is still under debate, but for now
+ * this is what we have settled on. Future changes should include 'funding',
+ * 'defunding', and state related to challenging, probably. See this issue
+ * for more: https://github.com/statechannels/statechannels/issues/2509
  */
-export const stage = (state: State | undefined): Stage =>
-  !state
-    ? 'Missing'
-    : state.isFinal
-    ? 'Final'
-    : state.turnNum <= state.participants.length - 1
-    ? 'PrefundSetup'
-    : state.turnNum <= state.participants.length * 2 - 1
-    ? 'PostfundSetup'
-    : 'Running';
+export const status = (channelState: ChannelState): ChannelStatus => {
+  const {supported, latest, latestSignedByMe, support} = channelState;
+  const {participants} = supported ?? latest;
+  if (latest.isFinal) {
+    if (support?.every(s => s.isFinal)) {
+      return 'closed'; // the entire support chain isFinal
+    } else {
+      return 'closing'; // at least one isFinal state proposed
```

Comment:
This might be incorrect. It could be that something in the `support` is `isFinal`, but `latest` is not `isFinal`. Is there another bit of code that guarantees this cannot happen?

Reply:
Is that actually possible? [It shouldn't be possible to transition from an `isFinal` state to a state that is not `isFinal`](https://github.com/statechannels/statechannels/blob/357a488076e4129f5029093412c0a9062d358d4d/packages/nitro-protocol/contracts/ForceMove.sol#L517-L520), so any support with an `isFinal` state must have `latest.isFinal`?

Reply:
Don't we store unsupported states? What stops Bob from sending Alice a later state with `isFinal: false`?